client/packages/common/src/ui/layout/tables/material-react-table/useTableDisplayOptions.tsx

```diff
@@ -210,61 +210,67 @@ export const useTableDisplayOptions = <T extends MRT_RowData>({
       },
     }),
 
-    muiTableBodyCellProps: ({ row, column, table }) => ({
-      sx: {
-        fontSize: table.getState().density === 'compact' ? '0.90em' : '1em',
-        fontWeight: 400,
-        opacity: 1,
-        color: getIsPlaceholderRow(row.original)
-          ? 'secondary.light'
-          : getIsRestrictedRow(row.original)
-            ? 'gray.main'
-            : undefined,
+    muiTableBodyCellProps: ({ row, column, table }) => {
+      const columnWidth = column.getSize();
+      return {
+        sx: {
+          fontSize: table.getState().density === 'compact' ? '0.90em' : '1em',
+          fontWeight: 400,
+          opacity: 1,
+          color: getIsPlaceholderRow(row.original)
+            ? 'secondary.light'
+            : getIsRestrictedRow(row.original)
+              ? 'gray.main'
+              : undefined,
 
-        ...(column.id === 'mrt-row-expand' && {
-          // The expand chevron is rotated incorrectly by default (in terms of
-          // consistency with other Accordion/Expando UI elements in the app)
-          button: {
-            rotate: row.getIsExpanded() ? '180deg' : '-90deg',
-            // Height and padding affect the density of the row
-            padding: 0,
-            height: 'unset',
-          },
-          // Hide the icon when there's nothing to expand
-          '& button.Mui-disabled': {
-            color: !row.getCanExpand() ? 'transparent' : undefined,
-          },
-        }),
-        minHeight: table.getState().density === 'compact' ? '32px' : '40px',
-        padding:
-          table.getState().density === 'spacious'
-            ? '0.7rem'
-            : table.getState().density === 'comfortable'
-              ? '0.35rem 0.5rem'
-              : undefined, // default for "compact",
+          ...(column.id === 'mrt-row-expand' && {
+            // The expand chevron is rotated incorrectly by default (in terms of
+            // consistency with other Accordion/Expando UI elements in the app)
+            button: {
+              rotate: row.getIsExpanded() ? '180deg' : '-90deg',
+              // Height and padding affect the density of the row
+              padding: 0,
+              height: 'unset',
+            },
+            // Hide the icon when there's nothing to expand
+            '& button.Mui-disabled': {
+              color: !row.getCanExpand() ? 'transparent' : undefined,
+            },
+          }),
+          minHeight: table.getState().density === 'compact' ? '32px' : '40px',
+          padding:
+            table.getState().density === 'spacious'
+              ? '0.7rem'
+              : table.getState().density === 'comfortable'
+                ? columnWidth > 100
+                  ? '0.35rem 0.5rem'
+                  : // Reduce the padding when column is narrow
+                    '0.35rem 0.25rem'
+                : undefined, // default for "compact",
```

Comment:
This is the actual change here, the rest is whitespace due to the indent at the top (to get the columnWidth). I added this because the UnitAndDose cell columns were wrapping even with values of 0 when they should have had plenty of width. So this condition reduces the cell padding when the column is narrow.